doc/design.md

```diff
@@ -1,51 +1,203 @@
+
+For the files in the `require_exact` PK also validates that the content is exactly the same.
+
+Rationale:
+This way of defining the project structure makes it very convenient to add new content. No knowledge about the internal implementation of this plugin is required. That allows users to quickly add or modify the template.
+
+An alternative would be to define the files by Java classes. In order to contain the contents of the required files, these classes would contain many String concatenations which would be very unreadable.
+
+Covers:
+
+* `req~verify-existence-of-files~1`
+
+Needs: impl, utest, itest
+
+### Deleted Files Validator
+
+`dsn~deleted-files-validator~1`
+
+For configuring the deleted files validator we decided for a Java API. For each deleted file one can specify a reason that will be printed as error message. By that one can for example tell that the file was moved.
+
+Covers:
+
+* `req~verify-non-existence-of-files~1`
+
+Needs: impl, utest, itest
+
+### Maven Plugin Validator
+
+`dsn~mvn-plugin-validator~1`
+
+We decided to implement the verification of the pom plugins using a combination of xml files and Java code. The XML files (in resources/mvn_templates) contain the XML of a typical plugin definition. The Java class defines which XML properties must be like define in the template.
+
+This approach allows us to quickly add new templates by copying an exemplary plugin definition from a pom.xml file, and the defining which parts are mandatory using Java.
+
+Covers:
+
+* `req~verify-mvn-plugins~1`
+
+Needs: impl, utest, itest
+
+### Mvn Dependency Validator
+
+`dsn~mvn-dependency-validator~1`
+
+We implemented the validation for dependency declaration using Java code. That allows us to add specific validations for certain dependencies. For example for the jacoco dependency we need to validate the scope.
+
+However, implementing it purely via configuration would probably also be a good solution. In the future maybe even a better one, depending on how many dependencies should get validated.
+
+Covers:
+
+* `req~verify-maven-dependencies~1`
+
+Needs: impl, utest, itest
+
+### Dependency Changes Section in changes_x.x.x.md File Validator
+
+`dsn~dependency-section-in-changes_x.x.x.md-file-validator~1`
+
+The basic approach for this validator is to generate the markdown for the `Dependency Changes` section and then compare the content of the project's file with that generated content. In case they differ the validation fails. The automatic fix of the validation error is to override the section with the generated one.
+
+For this feature we need to generate the list of dependency changes since the last release. We decided to generate this list by comparing the dependencies of the last release with the dependencies of this release.
```

Comment:
```suggestion
For this feature we need to generate the list of dependency changes since the last release. We decided to generate this list by comparing the dependencies of the last release with the dependencies of the current release.
```